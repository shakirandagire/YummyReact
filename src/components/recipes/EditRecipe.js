import React, { Component } from 'react';
import axios from 'axios';
import { notify } from 'react-notify-toast';
import { BASE_URL } from '../../../src/constants';
import Navigation from '../Navigation';

/**
 * Component for editing recipes
 */

class EditRecipe extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipename: '',
      recipe_description: '',
      instructions: '',
    };
  }
  // Function that populates the edit recipes form
  componentWillMount() {
    const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
    const catId = this.props.match.params.category_id;
    const recipeId = this.props.match.params.id;
    axios
      .get(`${BASE_URL}/api/v1/categories/${catId}/recipes/${recipeId}`, { headers })
      .then((response) => {
        this.setState({
          recipename: response.data.recipename,
          recipe_description: response.data.recipe_description,
          instructions: response.data.instructions,
        });
      })
      .catch((error) => {
        if (error.response) {
          notify.show(error.response.data.message);
        } else if (error.request) {
          notify.show("Request not made");
        }
      });
  }

  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }
  // Function that handles the editting of recipes

  handleEditRecipe = (event) => {
    const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
    event.preventDefault();
    const catId = this.props.match.params.category_id;
    const recipeId = this.props.match.params.id;
    axios
      .put(`${BASE_URL}/api/v1/categories/${catId}/recipes/${recipeId}`, this.state, { headers })
      .then(() => {
        notify.show("Recipe edited successfully", 'success', 4000);
        this.props.history.push(`/view-recipes/${catId}/recipes`);
      })
      .catch((error) => {
        if (error.response) {
          notify.show(error.response.data.message, 'success', 4000);
        } else if (error.request) {
          notify.show("Request not made");
        }
      });
  }

  render() {
    return (
      <div className="my_container backgimg">
        <Navigation />
        <div className="mycontent">

          <form onSubmit={this.handleEditRecipe}>

            <div className="form-group">
              Recipe Name:
              <input type="text" name="recipename" className="form-control" required onChange={this.handleInputChange} value={this.state.recipename} />
            </div>
            <div className="form-group">
              Description:
              <input type="text-area" name="recipe_description" className="form-control" required onChange={this.handleInputChange} value={this.state.recipe_description} />
            </div>
            <div className="form-group">
              Instructions:
              <input type="text-area" name="instructions" className="form-control" required onChange={this.handleInputChange} value={this.state.instructions} />
            </div>

            <button type="submit" className="btn btn-danger formsave" >Save</button>
          </form>
        </div>

      </div>


    );
  }
}
export default EditRecipe;
