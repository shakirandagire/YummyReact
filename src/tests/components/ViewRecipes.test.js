import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ViewRecipes, { Recipe } from '../../components/recipes/ViewRecipe';
import EditRecipe from '../../components/recipes/EditRecipe';

describe('<ViewRecipes/>', () => {
  const params = {
    match: {
      params: {
        category_id: 1,
      },
    },
  };

  const preventDefault = jest.fn();
  const component = shallow(<ViewRecipes match={{ params }} />);
  it('should render view recipes', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
//   it('should have get recipes method', () => {
//     expect(component.instance().getRecipes({ preventDefault }));
//   });
//   it('should have delete recipe method', () => {
//     expect(component.instance().deleteRecipes({ preventDefault }));
//   });
//   it('should have handle page method', () => {
//     expect(component.instance().handlePage({ preventDefault }));
//   });

//   it('should have initial state', () => {
//     expect(component.state().recipes).toEqual([]);
//     expect(component.state().q).toEqual("");
//     expect(component.state().perPage).toEqual(6);
//     expect(component.state().total).toEqual("");
//   });
});

describe('<Recipe />', () => {
  const params = {
    match: {
      params: {
        category_id: 1,
      },
    },
  };

  const preventDefault = jest.fn();
  const component = shallow(<Recipe />);
  it('should render view recipes', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  // it('should edit on click', () => {
  //   const wrapper = shallow(<EditRecipe match={{ params }} />);
  //   component.find('.btn-success .edit').simulate('click');
  //   expect(wrapper.instance().handleEditRecipe({ preventDefault }));
  // });
});

