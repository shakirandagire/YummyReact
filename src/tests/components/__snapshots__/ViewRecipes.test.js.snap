// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewRecipes/> should render view recipes 1`] = `
<div
  className="my_container backgimg"
>
  <Navigation />
  <div
    className="mycontent"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        Search for a recipe:
        <input
          className="form-control"
          name="q"
          onChange={[Function]}
          placeholder="search"
          type="text"
          value=""
        />
        <button
          className="btn btn-danger formsave"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
    <h3>
       Recipes 
    </h3>
    <button
      className="btn btn-default btn-md"
      type="button"
    >
      <Link
        replace={false}
        to="/add-recipes/undefined/recipes"
      >
        <span
          className="glyphicon glyphicon-trash"
        />
         Add recipes
      </Link>
    </button>
    <div
      className="row"
    />
    <Pagination
      handlePage={[Function]}
      perPage={6}
      total=""
    />
  </div>
</div>
`;
